clickanalysis/middleware/__init__.py

```diff
@@ -10,32 +13,22 @@ class ClickAnalysisMiddleware(object):
     """
 
     def process_request(self, request):
+
         if request.GET.get('cid', ''):  # Will process the logic only if campaign id is present
             try:
                 campaign = Campaign.objects.get(campaign_id=request.GET['cid'])
-            except:
+            except Campaign.DoesNotExist:
                 campaign = None
 
             # Clearing the cid parameter from the link or providing a clean url for user to redirect
-            get_params = request.get_full_path().split('?')[1].split('&')
-            params_to_send = []
-            for g_p in get_params:
-                name = g_p.split('=')[0]
-                if name != 'cid':
-                    params_to_send.append(g_p)
-
-            params = '&'.join(params_to_send)
-            if params_to_send:
-                url = request.get_full_path().split('?')[0] + '?' + params
-            else:
-                url = request.get_full_path().split('?')[0]
-
-            if campaign:  # If campaign was found then here the click will be recorded
+            url_parsed = urlparse(request.get_full_path())
+            get_params = parse_qsl(url_parsed.query)
+            params_to_send = urlencode([(k, v) for k, v in get_params if k != 'cid'])
+            url = urlunsplit(url_parsed[:3] + (params_to_send,) + url_parsed[5:])
+
+            if campaign is not None:  # If campaign was found then here the click will be recorded
```

Comment:
Logically yes, they are exactly the same. As we're not doing anything *special* in that if-else condition I reduced them to one line(lesser bytecodes :-)) that does the same thing. But if you want to stick with your version then it's okay as well because it is much more **explicit**.